devel/jenkins/Portfile

```diff
@@ -0,0 +1,96 @@
+
+PortSystem          1.0
+PortGroup           java 1.0
+
+name                jenkins
+version             2.401.3
+revision            0
+
+homepage            https://jenkins.io
+
+description         The leading open source automation server,\
+                    Jenkins provides hundreds of plugins to support building,\
+                    deploying and automating any project.
+
+long_description    ${description}
+
+categories          devel java
+license             MIT
+
+maintainers         {gmail.com:slashapp @slashapp} openmaintainer
+
+master_sites        https://get.jenkins.io/war-stable/${version}/
+distfiles           ${name}.war
+
+checksums           rmd160  0a762953129148086beced9450a364467a136b40 \
+                    sha256  a798a0c5481a8ffb0320d9121f6cf49dc575c369028daae17a4dd398b69e000d \
+                    size    98398119
+
+java.version        11+
+java.fallback       openjdk11
+
+use_configure       no
+
+extract {}
+
+build {}
+
+set jenkins_launch_dir  /Library/LaunchDaemons
+set jenkins_plist_dir   ${prefix}/etc/LaunchDaemons/org.macports.${name}
+set jenkins_plist       org.macports.${name}.plist
+set jenkins_home_dir    ${prefix}/var/lib/${name}
+set jenkins_exec_dir    ${prefix}/libexec/${name}
+set jenkins_exec        ${jenkins_exec_dir}/${distfiles}
+set jenkins_host        127.0.0.1
+set jenkins_port        8080
+
+post-patch {
+    copy ${filespath}/${jenkins_plist} ${workpath}
+    reinplace "s|@JENKINS_HOME@|${jenkins_home_dir}|g"  ${workpath}/${jenkins_plist}
+    reinplace "s|@JENKINS_EXEC@|${jenkins_exec}|g"      ${workpath}/${jenkins_plist}
+    reinplace "s|@JENKINS_HOST@|${jenkins_host}|g"      ${workpath}/${jenkins_plist}
+    reinplace "s|@JENKINS_PORT@|${jenkins_port}|g"      ${workpath}/${jenkins_plist}
+}
+
+destroot.keepdirs-append \
+    ${destroot}${jenkins_plist_dir} \
+    ${destroot}${jenkins_home_dir} \
+    ${destroot}${jenkins_exec_dir}
+
+destroot {
+    xinstall -d -m 0755 ${destroot}${jenkins_plist_dir}
+    xinstall -d -m 0755 ${destroot}${jenkins_home_dir}
+    xinstall -d -m 0755 ${destroot}${jenkins_exec_dir}
+    xinstall -d -m 0755 ${destroot}${jenkins_launch_dir}
+
+    xinstall -m 0644 -o root -W ${workpath} ${jenkins_plist} \
+        ${destroot}${jenkins_plist_dir}
+
+    ln -s ${jenkins_plist_dir}/${jenkins_plist} \
+        ${destroot}${jenkins_launch_dir}/${jenkins_plist}
+
+    copy ${distpath}/${distfiles}  ${destroot}${jenkins_exec}
+}
+
+notes "
+To configure Jenkins service, edit plist as follows:
+
+  Read document at https://jenkins.io/doc/book/installing/initial-settings/
+
+  \$ sudo vi ${jenkins_launch_dir}/${jenkins_plist}
+
+To enable the Jenkins service, use `port load`, as follows:
+
+\$ sudo port load ${name}
+
+Once the service is enabled, Jenkins will:
+
+  - listen by default on: http://localhost:8080
+
+Here is the default Administrator password:
+\$ sudo cat ${jenkins_home_dir}/.jenkins/secrets/initialAdminPassword
```

Comment:
Is this file supposed to be installed by the port?

Testing locally, this is what I'm seeing:

```
$ port contents jenkins
  /Library/LaunchDaemons/org.macports.jenkins.plist
  /opt/local/etc/LaunchDaemons/org.macports.jenkins/.turd_jenkins
  /opt/local/etc/LaunchDaemons/org.macports.jenkins/org.macports.jenkins.plist
  /opt/local/libexec/jenkins/.turd_jenkins
  /opt/local/libexec/jenkins/jenkins.war
  /opt/local/var/lib/jenkins/.turd_jenkins
```

Reply:
It's a password file generated on first startup, the port showing it at welcome page and can be removed from the notes.